account_move_template/tests/test_account_move_template_options.py

```diff
@@ -0,0 +1,294 @@
+            )
+        template_run = f.save()
+        msg_error = re.escape(
+            "Valid fields to overwrite are ['partner_id', 'amount', 'name', 'date_maturity']"
+        )
+        with self.assertRaisesRegex(ValidationError, msg_error):
+            template_run.load_lines()
+        # Assign only on valid fields, and load_lines again
+        with Form(self.env["account.move.template.run"]) as f:
+            f.template_id = self.move_template
+            f.overwrite = str(
+                {
+                    "L0": {"partner_id": self.partners[0].id, "amount": 3000},
+                    "L1": {"partner_id": self.partners[1].id},
+                    "L2": {"partner_id": self.partners[2].id},
+                }
+            )
+        template_run = f.save()
+        res = template_run.load_lines()
+        self.assertEqual(template_run.line_ids[0].partner_id, self.partners[0])
+        self.assertEqual(template_run.line_ids[0].amount, 3000)
+        res = template_run.with_context(**res["context"]).generate_move()
+        move = self.Move.browse(res["res_id"])
+        self.assertRecordValues(
+            move.line_ids.sorted("credit"),
+            [
+                {
+                    "partner_id": self.partners[0].id,
+                    "account_id": self.ar_account_id.id,
+                    "credit": 0.0,
+                    "debit": 3000.0,
+                },
+                {
+                    "partner_id": self.partners[1].id,
+                    "account_id": self.income_account_id.id,
+                    "credit": 1000.0,
+                    "debit": 0.0,
+                },
+                {
+                    "partner_id": self.partners[2].id,
+                    "account_id": self.income_account_id.id,
+                    "credit": 2000.0,
+                    "debit": 0.0,
+                },
+            ],
+        )
+
+    def test_move_copy(self):
+        template_copy = self.move_template.copy()
+        self.assertEqual(template_copy.name, "Test Template (copy)")
+
+    def test_move_generate_from_action_button(self):
+        # `Generate Journal Entry` action button
+        res = self.move_template.generate_journal_entry()
+        self.assertEqual(res["name"], "Create Entry from Template")
+        self.assertEqual(res["res_model"], "account.move.template.run")
+
+    def test_move_template_exceptions(self):
+        msg_error = "Python Code must be set for computed line with sequence 1."
+        with self.assertRaisesRegex(ValidationError, msg_error):
+            self.move_template.line_ids[1].python_code = ""
+
+        self.move_template.line_ids[1].python_code = "P0*1/3"
+        with Form(self.env["account.move.template.run"]) as f:
+            f.template_id = self.move_template
+            template_run = f.save()
+        template_run.load_lines()
+        msg_error = re.escape(
+            "Impossible to compute the formula of line with sequence 1 "
```

Comment:
FYI you don't need the whole error message, only a fragment long enough to know this is the expected error. In this case, last line would be enough IMHO.

Same below.

In addition, using that way, you won't need to use `re.escape`